exchanges/kraken/kraken_test.go

```diff
@@ -2102,3 +2116,84 @@ func TestIsPerpetualFutureCurrency(t *testing.T) {
 		t.Error("expected true")
 	}
 }
+
+func testInstance() (*Kraken, error) {
+	k := new(Kraken)
+	k.SetDefaults()
+	cfg := config.GetConfig()
+	err := cfg.LoadConfig("../../testdata/configtest.json", true)
+	if err != nil {
+		return nil, fmt.Errorf("Error loading config: %w", err)
+	}
+	krakenConfig, err := cfg.GetExchangeConfig("Kraken")
+	if err != nil {
+		return nil, fmt.Errorf("Error getting exchange config: %w", err)
+	}
+	if apiKey != "" {
+		krakenConfig.API.Credentials.Key = apiKey
+	}
+	if apiSecret != "" {
+		krakenConfig.API.Credentials.Secret = apiSecret
+	}
+	k.Websocket = sharedtestvalues.NewTestWebsocket()
+	err = k.Setup(krakenConfig)
+
+	return k, err
+}
+
+var upgrader = websocket.Upgrader{}
```

Comment:
can rm and inline in `wsMockWrapper` below

Reply:
Agreed.
I didn't think at all about why examples tended to have it as a global.

Fixed 9ee6596f